File: go/day81.go
package main

import (
	"fmt"
	"bufio"
	"os"
	"strings"
	"regexp"
)

func main() {
	fmt.Println("Please input your card number")


	reader := bufio.NewReader(os.Stdin)

	cardNumber,_ := reader.ReadString('\n')

	cardNumber = strings.TrimSpace(cardNumber)

	match,_ := regexp.MatchString("^\\d{16}$",cardNumber)


	if match {
		fmt.Println("Valid card number!")
	} else {
		fmt.Println("Invalid card number!!!")
	}
}

File: dart/day81.dart
import "dart:io";
  
void main() {
  stdout.write('Enter your card number: '); // Prompt the user
  String? cardNumber = stdin.readLineSync();

  if (cardNumber == null) {
  stdout.write('Failed');
    return;
  }
  
  RegExp cardRegex = RegExp("^[0-9]{16}\$");
  
  if (cardRegex.hasMatch(cardNumber)) {
    stdout.write('Card number is valid');
  }
  else {
    stdout.write('Card number is invalid!!!');
  }
  
}